shub/deploy.py

```diff
@@ -140,17 +141,17 @@ def _upload_egg(target, eggpath, project, version, auth):
     return make_deploy_request(url, data, files, auth)
 
 
-def _build_egg():
+def _build_egg(setup_file):
     closest = closest_scrapy_cfg()
     os.chdir(os.path.dirname(closest))
-    if not os.path.exists('setup.py'):
+    if not os.path.exists(setup_file):
```

Comment:
Yes, I still think that throwing an error is the way to go for user defined setup files.